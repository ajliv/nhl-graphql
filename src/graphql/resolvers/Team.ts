import { GraphQLFieldResolver } from 'graphql';
import { sortBy } from 'lodash';

import { Context } from '..';

export const Team: Record<
    string,
    GraphQLFieldResolver<Record<string, any>, Context>
> = {
    conference({ conference }, _, { dataSources }) {
        return dataSources.nhl.getConference(conference?.id);
    },

    division({ division }, _, { dataSources }) {
        return dataSources.nhl.getDivision(division?.id);
    },

    roster({ roster }, _, { dataSources }) {
        const items = sortBy(roster?.roster ?? [], r =>
            parseInt(r.jerseyNumber, 10)
        );
        return Promise.all(
            items.map(r => dataSources.nhl.getPerson(r.person.id))
        );
    },
};
