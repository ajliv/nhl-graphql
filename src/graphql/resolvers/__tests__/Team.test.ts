import { Team } from '../Team';
import { mockTeam, mockNHL } from '../../datasources/__tests__/fixtures';

const nhl = mockNHL();

const mockContext = {
    now: new Date(1569902400000), // 2019-10-01T00:00:00.000Z,
    dataSources: {
        nhl,
    },
};

it('should fetch a roster', async () => {
    const { roster } = Team as any;
    const mockRoster = mockTeam.roster.roster;

    const resp = await roster(mockTeam, {}, mockContext);

    expect(nhl.getPerson).toHaveBeenCalledTimes(mockRoster.length);

    for (const { person } of mockRoster) {
        expect(nhl.getPerson).toHaveBeenCalledWith(person.id);
    }

    expect(resp).toMatchSnapshot();
});
