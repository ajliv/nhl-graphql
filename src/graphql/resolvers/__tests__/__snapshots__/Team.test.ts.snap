// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should fetch a roster 1`] = `
Array [
  Object {
    "active": true,
    "alternateCaptain": false,
    "birthCity": "Roubaix",
    "birthCountry": "FRA",
    "birthDate": "1989-11-21",
    "captain": false,
    "currentAge": 30,
    "currentTeam": Object {
      "id": 23,
      "link": "/api/v1/teams/23",
      "name": "Vancouver Canucks",
    },
    "firstName": "Antoine",
    "fullName": "Antoine Roussel",
    "height": "5' 11\\"",
    "id": 8474849,
    "lastName": "Roussel",
    "link": "/api/v1/people/8474849",
    "nationality": "FRA",
    "primaryNumber": "26",
    "primaryPosition": Object {
      "abbreviation": "LW",
      "code": "L",
      "name": "Left Wing",
      "type": "Forward",
    },
    "rookie": false,
    "rosterStatus": "Y",
    "shootsCatches": "L",
    "weight": 199,
  },
  Object {
    "active": true,
    "alternateCaptain": false,
    "birthCity": "Roubaix",
    "birthCountry": "FRA",
    "birthDate": "1989-11-21",
    "captain": false,
    "currentAge": 30,
    "currentTeam": Object {
      "id": 23,
      "link": "/api/v1/teams/23",
      "name": "Vancouver Canucks",
    },
    "firstName": "Antoine",
    "fullName": "Antoine Roussel",
    "height": "5' 11\\"",
    "id": 8474849,
    "lastName": "Roussel",
    "link": "/api/v1/people/8474849",
    "nationality": "FRA",
    "primaryNumber": "26",
    "primaryPosition": Object {
      "abbreviation": "LW",
      "code": "L",
      "name": "Left Wing",
      "type": "Forward",
    },
    "rookie": false,
    "rosterStatus": "Y",
    "shootsCatches": "L",
    "weight": 199,
  },
  Object {
    "active": true,
    "alternateCaptain": false,
    "birthCity": "Roubaix",
    "birthCountry": "FRA",
    "birthDate": "1989-11-21",
    "captain": false,
    "currentAge": 30,
    "currentTeam": Object {
      "id": 23,
      "link": "/api/v1/teams/23",
      "name": "Vancouver Canucks",
    },
    "firstName": "Antoine",
    "fullName": "Antoine Roussel",
    "height": "5' 11\\"",
    "id": 8474849,
    "lastName": "Roussel",
    "link": "/api/v1/people/8474849",
    "nationality": "FRA",
    "primaryNumber": "26",
    "primaryPosition": Object {
      "abbreviation": "LW",
      "code": "L",
      "name": "Left Wing",
      "type": "Forward",
    },
    "rookie": false,
    "rosterStatus": "Y",
    "shootsCatches": "L",
    "weight": 199,
  },
  Object {
    "active": true,
    "alternateCaptain": false,
    "birthCity": "Roubaix",
    "birthCountry": "FRA",
    "birthDate": "1989-11-21",
    "captain": false,
    "currentAge": 30,
    "currentTeam": Object {
      "id": 23,
      "link": "/api/v1/teams/23",
      "name": "Vancouver Canucks",
    },
    "firstName": "Antoine",
    "fullName": "Antoine Roussel",
    "height": "5' 11\\"",
    "id": 8474849,
    "lastName": "Roussel",
    "link": "/api/v1/people/8474849",
    "nationality": "FRA",
    "primaryNumber": "26",
    "primaryPosition": Object {
      "abbreviation": "LW",
      "code": "L",
      "name": "Left Wing",
      "type": "Forward",
    },
    "rookie": false,
    "rosterStatus": "Y",
    "shootsCatches": "L",
    "weight": 199,
  },
  Object {
    "active": true,
    "alternateCaptain": false,
    "birthCity": "Roubaix",
    "birthCountry": "FRA",
    "birthDate": "1989-11-21",
    "captain": false,
    "currentAge": 30,
    "currentTeam": Object {
      "id": 23,
      "link": "/api/v1/teams/23",
      "name": "Vancouver Canucks",
    },
    "firstName": "Antoine",
    "fullName": "Antoine Roussel",
    "height": "5' 11\\"",
    "id": 8474849,
    "lastName": "Roussel",
    "link": "/api/v1/people/8474849",
    "nationality": "FRA",
    "primaryNumber": "26",
    "primaryPosition": Object {
      "abbreviation": "LW",
      "code": "L",
      "name": "Left Wing",
      "type": "Forward",
    },
    "rookie": false,
    "rosterStatus": "Y",
    "shootsCatches": "L",
    "weight": 199,
  },
  Object {
    "active": true,
    "alternateCaptain": false,
    "birthCity": "Roubaix",
    "birthCountry": "FRA",
    "birthDate": "1989-11-21",
    "captain": false,
    "currentAge": 30,
    "currentTeam": Object {
      "id": 23,
      "link": "/api/v1/teams/23",
      "name": "Vancouver Canucks",
    },
    "firstName": "Antoine",
    "fullName": "Antoine Roussel",
    "height": "5' 11\\"",
    "id": 8474849,
    "lastName": "Roussel",
    "link": "/api/v1/people/8474849",
    "nationality": "FRA",
    "primaryNumber": "26",
    "primaryPosition": Object {
      "abbreviation": "LW",
      "code": "L",
      "name": "Left Wing",
      "type": "Forward",
    },
    "rookie": false,
    "rosterStatus": "Y",
    "shootsCatches": "L",
    "weight": 199,
  },
  Object {
    "active": true,
    "alternateCaptain": false,
    "birthCity": "Roubaix",
    "birthCountry": "FRA",
    "birthDate": "1989-11-21",
    "captain": false,
    "currentAge": 30,
    "currentTeam": Object {
      "id": 23,
      "link": "/api/v1/teams/23",
      "name": "Vancouver Canucks",
    },
    "firstName": "Antoine",
    "fullName": "Antoine Roussel",
    "height": "5' 11\\"",
    "id": 8474849,
    "lastName": "Roussel",
    "link": "/api/v1/people/8474849",
    "nationality": "FRA",
    "primaryNumber": "26",
    "primaryPosition": Object {
      "abbreviation": "LW",
      "code": "L",
      "name": "Left Wing",
      "type": "Forward",
    },
    "rookie": false,
    "rosterStatus": "Y",
    "shootsCatches": "L",
    "weight": 199,
  },
  Object {
    "active": true,
    "alternateCaptain": false,
    "birthCity": "Roubaix",
    "birthCountry": "FRA",
    "birthDate": "1989-11-21",
    "captain": false,
    "currentAge": 30,
    "currentTeam": Object {
      "id": 23,
      "link": "/api/v1/teams/23",
      "name": "Vancouver Canucks",
    },
    "firstName": "Antoine",
    "fullName": "Antoine Roussel",
    "height": "5' 11\\"",
    "id": 8474849,
    "lastName": "Roussel",
    "link": "/api/v1/people/8474849",
    "nationality": "FRA",
    "primaryNumber": "26",
    "primaryPosition": Object {
      "abbreviation": "LW",
      "code": "L",
      "name": "Left Wing",
      "type": "Forward",
    },
    "rookie": false,
    "rosterStatus": "Y",
    "shootsCatches": "L",
    "weight": 199,
  },
  Object {
    "active": true,
    "alternateCaptain": false,
    "birthCity": "Roubaix",
    "birthCountry": "FRA",
    "birthDate": "1989-11-21",
    "captain": false,
    "currentAge": 30,
    "currentTeam": Object {
      "id": 23,
      "link": "/api/v1/teams/23",
      "name": "Vancouver Canucks",
    },
    "firstName": "Antoine",
    "fullName": "Antoine Roussel",
    "height": "5' 11\\"",
    "id": 8474849,
    "lastName": "Roussel",
    "link": "/api/v1/people/8474849",
    "nationality": "FRA",
    "primaryNumber": "26",
    "primaryPosition": Object {
      "abbreviation": "LW",
      "code": "L",
      "name": "Left Wing",
      "type": "Forward",
    },
    "rookie": false,
    "rosterStatus": "Y",
    "shootsCatches": "L",
    "weight": 199,
  },
  Object {
    "active": true,
    "alternateCaptain": false,
    "birthCity": "Roubaix",
    "birthCountry": "FRA",
    "birthDate": "1989-11-21",
    "captain": false,
    "currentAge": 30,
    "currentTeam": Object {
      "id": 23,
      "link": "/api/v1/teams/23",
      "name": "Vancouver Canucks",
    },
    "firstName": "Antoine",
    "fullName": "Antoine Roussel",
    "height": "5' 11\\"",
    "id": 8474849,
    "lastName": "Roussel",
    "link": "/api/v1/people/8474849",
    "nationality": "FRA",
    "primaryNumber": "26",
    "primaryPosition": Object {
      "abbreviation": "LW",
      "code": "L",
      "name": "Left Wing",
      "type": "Forward",
    },
    "rookie": false,
    "rosterStatus": "Y",
    "shootsCatches": "L",
    "weight": 199,
  },
  Object {
    "active": true,
    "alternateCaptain": false,
    "birthCity": "Roubaix",
    "birthCountry": "FRA",
    "birthDate": "1989-11-21",
    "captain": false,
    "currentAge": 30,
    "currentTeam": Object {
      "id": 23,
      "link": "/api/v1/teams/23",
      "name": "Vancouver Canucks",
    },
    "firstName": "Antoine",
    "fullName": "Antoine Roussel",
    "height": "5' 11\\"",
    "id": 8474849,
    "lastName": "Roussel",
    "link": "/api/v1/people/8474849",
    "nationality": "FRA",
    "primaryNumber": "26",
    "primaryPosition": Object {
      "abbreviation": "LW",
      "code": "L",
      "name": "Left Wing",
      "type": "Forward",
    },
    "rookie": false,
    "rosterStatus": "Y",
    "shootsCatches": "L",
    "weight": 199,
  },
  Object {
    "active": true,
    "alternateCaptain": false,
    "birthCity": "Roubaix",
    "birthCountry": "FRA",
    "birthDate": "1989-11-21",
    "captain": false,
    "currentAge": 30,
    "currentTeam": Object {
      "id": 23,
      "link": "/api/v1/teams/23",
      "name": "Vancouver Canucks",
    },
    "firstName": "Antoine",
    "fullName": "Antoine Roussel",
    "height": "5' 11\\"",
    "id": 8474849,
    "lastName": "Roussel",
    "link": "/api/v1/people/8474849",
    "nationality": "FRA",
    "primaryNumber": "26",
    "primaryPosition": Object {
      "abbreviation": "LW",
      "code": "L",
      "name": "Left Wing",
      "type": "Forward",
    },
    "rookie": false,
    "rosterStatus": "Y",
    "shootsCatches": "L",
    "weight": 199,
  },
  Object {
    "active": true,
    "alternateCaptain": false,
    "birthCity": "Roubaix",
    "birthCountry": "FRA",
    "birthDate": "1989-11-21",
    "captain": false,
    "currentAge": 30,
    "currentTeam": Object {
      "id": 23,
      "link": "/api/v1/teams/23",
      "name": "Vancouver Canucks",
    },
    "firstName": "Antoine",
    "fullName": "Antoine Roussel",
    "height": "5' 11\\"",
    "id": 8474849,
    "lastName": "Roussel",
    "link": "/api/v1/people/8474849",
    "nationality": "FRA",
    "primaryNumber": "26",
    "primaryPosition": Object {
      "abbreviation": "LW",
      "code": "L",
      "name": "Left Wing",
      "type": "Forward",
    },
    "rookie": false,
    "rosterStatus": "Y",
    "shootsCatches": "L",
    "weight": 199,
  },
  Object {
    "active": true,
    "alternateCaptain": false,
    "birthCity": "Roubaix",
    "birthCountry": "FRA",
    "birthDate": "1989-11-21",
    "captain": false,
    "currentAge": 30,
    "currentTeam": Object {
      "id": 23,
      "link": "/api/v1/teams/23",
      "name": "Vancouver Canucks",
    },
    "firstName": "Antoine",
    "fullName": "Antoine Roussel",
    "height": "5' 11\\"",
    "id": 8474849,
    "lastName": "Roussel",
    "link": "/api/v1/people/8474849",
    "nationality": "FRA",
    "primaryNumber": "26",
    "primaryPosition": Object {
      "abbreviation": "LW",
      "code": "L",
      "name": "Left Wing",
      "type": "Forward",
    },
    "rookie": false,
    "rosterStatus": "Y",
    "shootsCatches": "L",
    "weight": 199,
  },
  Object {
    "active": true,
    "alternateCaptain": false,
    "birthCity": "Roubaix",
    "birthCountry": "FRA",
    "birthDate": "1989-11-21",
    "captain": false,
    "currentAge": 30,
    "currentTeam": Object {
      "id": 23,
      "link": "/api/v1/teams/23",
      "name": "Vancouver Canucks",
    },
    "firstName": "Antoine",
    "fullName": "Antoine Roussel",
    "height": "5' 11\\"",
    "id": 8474849,
    "lastName": "Roussel",
    "link": "/api/v1/people/8474849",
    "nationality": "FRA",
    "primaryNumber": "26",
    "primaryPosition": Object {
      "abbreviation": "LW",
      "code": "L",
      "name": "Left Wing",
      "type": "Forward",
    },
    "rookie": false,
    "rosterStatus": "Y",
    "shootsCatches": "L",
    "weight": 199,
  },
  Object {
    "active": true,
    "alternateCaptain": false,
    "birthCity": "Roubaix",
    "birthCountry": "FRA",
    "birthDate": "1989-11-21",
    "captain": false,
    "currentAge": 30,
    "currentTeam": Object {
      "id": 23,
      "link": "/api/v1/teams/23",
      "name": "Vancouver Canucks",
    },
    "firstName": "Antoine",
    "fullName": "Antoine Roussel",
    "height": "5' 11\\"",
    "id": 8474849,
    "lastName": "Roussel",
    "link": "/api/v1/people/8474849",
    "nationality": "FRA",
    "primaryNumber": "26",
    "primaryPosition": Object {
      "abbreviation": "LW",
      "code": "L",
      "name": "Left Wing",
      "type": "Forward",
    },
    "rookie": false,
    "rosterStatus": "Y",
    "shootsCatches": "L",
    "weight": 199,
  },
  Object {
    "active": true,
    "alternateCaptain": false,
    "birthCity": "Roubaix",
    "birthCountry": "FRA",
    "birthDate": "1989-11-21",
    "captain": false,
    "currentAge": 30,
    "currentTeam": Object {
      "id": 23,
      "link": "/api/v1/teams/23",
      "name": "Vancouver Canucks",
    },
    "firstName": "Antoine",
    "fullName": "Antoine Roussel",
    "height": "5' 11\\"",
    "id": 8474849,
    "lastName": "Roussel",
    "link": "/api/v1/people/8474849",
    "nationality": "FRA",
    "primaryNumber": "26",
    "primaryPosition": Object {
      "abbreviation": "LW",
      "code": "L",
      "name": "Left Wing",
      "type": "Forward",
    },
    "rookie": false,
    "rosterStatus": "Y",
    "shootsCatches": "L",
    "weight": 199,
  },
  Object {
    "active": true,
    "alternateCaptain": false,
    "birthCity": "Roubaix",
    "birthCountry": "FRA",
    "birthDate": "1989-11-21",
    "captain": false,
    "currentAge": 30,
    "currentTeam": Object {
      "id": 23,
      "link": "/api/v1/teams/23",
      "name": "Vancouver Canucks",
    },
    "firstName": "Antoine",
    "fullName": "Antoine Roussel",
    "height": "5' 11\\"",
    "id": 8474849,
    "lastName": "Roussel",
    "link": "/api/v1/people/8474849",
    "nationality": "FRA",
    "primaryNumber": "26",
    "primaryPosition": Object {
      "abbreviation": "LW",
      "code": "L",
      "name": "Left Wing",
      "type": "Forward",
    },
    "rookie": false,
    "rosterStatus": "Y",
    "shootsCatches": "L",
    "weight": 199,
  },
  Object {
    "active": true,
    "alternateCaptain": false,
    "birthCity": "Roubaix",
    "birthCountry": "FRA",
    "birthDate": "1989-11-21",
    "captain": false,
    "currentAge": 30,
    "currentTeam": Object {
      "id": 23,
      "link": "/api/v1/teams/23",
      "name": "Vancouver Canucks",
    },
    "firstName": "Antoine",
    "fullName": "Antoine Roussel",
    "height": "5' 11\\"",
    "id": 8474849,
    "lastName": "Roussel",
    "link": "/api/v1/people/8474849",
    "nationality": "FRA",
    "primaryNumber": "26",
    "primaryPosition": Object {
      "abbreviation": "LW",
      "code": "L",
      "name": "Left Wing",
      "type": "Forward",
    },
    "rookie": false,
    "rosterStatus": "Y",
    "shootsCatches": "L",
    "weight": 199,
  },
  Object {
    "active": true,
    "alternateCaptain": false,
    "birthCity": "Roubaix",
    "birthCountry": "FRA",
    "birthDate": "1989-11-21",
    "captain": false,
    "currentAge": 30,
    "currentTeam": Object {
      "id": 23,
      "link": "/api/v1/teams/23",
      "name": "Vancouver Canucks",
    },
    "firstName": "Antoine",
    "fullName": "Antoine Roussel",
    "height": "5' 11\\"",
    "id": 8474849,
    "lastName": "Roussel",
    "link": "/api/v1/people/8474849",
    "nationality": "FRA",
    "primaryNumber": "26",
    "primaryPosition": Object {
      "abbreviation": "LW",
      "code": "L",
      "name": "Left Wing",
      "type": "Forward",
    },
    "rookie": false,
    "rosterStatus": "Y",
    "shootsCatches": "L",
    "weight": 199,
  },
  Object {
    "active": true,
    "alternateCaptain": false,
    "birthCity": "Roubaix",
    "birthCountry": "FRA",
    "birthDate": "1989-11-21",
    "captain": false,
    "currentAge": 30,
    "currentTeam": Object {
      "id": 23,
      "link": "/api/v1/teams/23",
      "name": "Vancouver Canucks",
    },
    "firstName": "Antoine",
    "fullName": "Antoine Roussel",
    "height": "5' 11\\"",
    "id": 8474849,
    "lastName": "Roussel",
    "link": "/api/v1/people/8474849",
    "nationality": "FRA",
    "primaryNumber": "26",
    "primaryPosition": Object {
      "abbreviation": "LW",
      "code": "L",
      "name": "Left Wing",
      "type": "Forward",
    },
    "rookie": false,
    "rosterStatus": "Y",
    "shootsCatches": "L",
    "weight": 199,
  },
  Object {
    "active": true,
    "alternateCaptain": false,
    "birthCity": "Roubaix",
    "birthCountry": "FRA",
    "birthDate": "1989-11-21",
    "captain": false,
    "currentAge": 30,
    "currentTeam": Object {
      "id": 23,
      "link": "/api/v1/teams/23",
      "name": "Vancouver Canucks",
    },
    "firstName": "Antoine",
    "fullName": "Antoine Roussel",
    "height": "5' 11\\"",
    "id": 8474849,
    "lastName": "Roussel",
    "link": "/api/v1/people/8474849",
    "nationality": "FRA",
    "primaryNumber": "26",
    "primaryPosition": Object {
      "abbreviation": "LW",
      "code": "L",
      "name": "Left Wing",
      "type": "Forward",
    },
    "rookie": false,
    "rosterStatus": "Y",
    "shootsCatches": "L",
    "weight": 199,
  },
  Object {
    "active": true,
    "alternateCaptain": false,
    "birthCity": "Roubaix",
    "birthCountry": "FRA",
    "birthDate": "1989-11-21",
    "captain": false,
    "currentAge": 30,
    "currentTeam": Object {
      "id": 23,
      "link": "/api/v1/teams/23",
      "name": "Vancouver Canucks",
    },
    "firstName": "Antoine",
    "fullName": "Antoine Roussel",
    "height": "5' 11\\"",
    "id": 8474849,
    "lastName": "Roussel",
    "link": "/api/v1/people/8474849",
    "nationality": "FRA",
    "primaryNumber": "26",
    "primaryPosition": Object {
      "abbreviation": "LW",
      "code": "L",
      "name": "Left Wing",
      "type": "Forward",
    },
    "rookie": false,
    "rosterStatus": "Y",
    "shootsCatches": "L",
    "weight": 199,
  },
  Object {
    "active": true,
    "alternateCaptain": false,
    "birthCity": "Roubaix",
    "birthCountry": "FRA",
    "birthDate": "1989-11-21",
    "captain": false,
    "currentAge": 30,
    "currentTeam": Object {
      "id": 23,
      "link": "/api/v1/teams/23",
      "name": "Vancouver Canucks",
    },
    "firstName": "Antoine",
    "fullName": "Antoine Roussel",
    "height": "5' 11\\"",
    "id": 8474849,
    "lastName": "Roussel",
    "link": "/api/v1/people/8474849",
    "nationality": "FRA",
    "primaryNumber": "26",
    "primaryPosition": Object {
      "abbreviation": "LW",
      "code": "L",
      "name": "Left Wing",
      "type": "Forward",
    },
    "rookie": false,
    "rosterStatus": "Y",
    "shootsCatches": "L",
    "weight": 199,
  },
]
`;
